File: src/main/java/com/yai/evaluator/SequenceEvaluator.java
package com.yai.evaluator;

import com.yai.YaInterpreterBaseVisitor;
import com.yai.YaInterpreterParser;
import com.yai.exception.SequenceEvaluatorException;
import com.yai.sequence.Sequence;
import org.antlr.v4.runtime.tree.ParseTree;

public class SequenceEvaluator extends YaInterpreterBaseVisitor<Sequence<Integer>> {
    private final Sequence<Integer> sequence = new Sequence<>();

    public SequenceEvaluator(ParseTree tree) {
        visit(tree);
    }

    public Sequence<Integer> getSequence() {
        return sequence;
    }

    @Override
    public Sequence<Integer> visitSequence(YaInterpreterParser.SequenceContext ctx) {
        if (!ctx.NUMBER().isEmpty() && ctx.NUMBER().size() == 2) {
            String fromStr = ctx.NUMBER().get(0).toString();
            String toStr = ctx.NUMBER().get(1).toString();
            if (isDouble(fromStr) || isDouble(toStr)) {
                throw new SequenceEvaluatorException("Sequence shouldn't have double");
            }
            int from = Integer.parseInt(fromStr);
            int to = Integer.parseInt(toStr);

            if (to > from ) {
                int n = to - from;
                for (int i = 0; i <= n; i++) {
                    sequence.add(from++);
                }
            }
        }
        return super.visitSequence(ctx);
    }

    private boolean isDouble(String str) {
        return str.equals(Double.toString(Double.parseDouble(str)));
    }

}

File: src/main/kotlin/com/yai/SequenceEvaluator.kt
//package com.yai
//
//import com.yai.YaInterpreterParser.SequenceContext
//import org.antlr.v4.runtime.tree.ParseTree
//
//
//class SequenceEvaluator(private val tree: ParseTree) : YaInterpreterBaseVisitor<List<Int>>() {
//    val sequence = listOf<Int>()
//
//    init {
//        visit(tree)
//    }
//
//    override fun visitStatement(ctx: YaInterpreterParser.StatementContext): List<Int> {
//        for (i in ctx.sequence().indices) {
//            visit(ctx.variable(i) as ParseTree)
//        }
//        return sequence
//    }
//
//    override fun visitSequence(ctx: SequenceContext): List<Int> {
//        if (ctx.INT().isNotEmpty() && ctx.INT().size == 2) {
//            var from = ctx.INT()[0].text.toInt()
//            var to = ctx.INT()[1].text.toInt()
//
//            if (to > from ) {
//                val n = to - from
//                for (i in 0..n) {
//                    sequence[from++]
//                }
//            }
//        }
//        return super.visitSequence(ctx)
//    }
//
//}